iree/turbine/kernel/wave/utils.py

```diff
@@ -419,14 +453,32 @@ def _print_bench_result(result, filename):
         print(res)
 
 
+def get_device_uuid(input_tensors: list[torch.Tensor]) -> tuple[int, str]:
+    """
+    Checks all torch.Tensor are on the same device, and get UUID from Torch device.
+    """
+    device_list = [
+        input.device for input in input_tensors if isinstance(input, torch.Tensor)
+    ]
+    if len(set(device_list)) != 1:
+        raise ValueError(f"Found multiple device on input tensors:{set(device_list)}")
+    device = device_list[0]
+    if device.type != "cuda":
+        raise ValueError("Expected all argument tensors to be in GPU.")
+    uuid = str(torch.cuda.get_device_properties(device).uuid)
+    return uuid
+
+
 def compile_and_invoke(
     asm: str,
     func_name: str,
     config: dict[str, str],
     kernel_inputs: list[torch.Tensor],
     kernel_outputs: list[torch.Tensor],
+    kernel_dynamic_dims: list[int] = [],
     run: bool = False,
     run_bench: bool = False,
+    inplace: bool = False,
```

Comment:
Shouldn't we default this to True? This seems like the preferred mode of execution. Do you see any downsides to turning this on by default?

Reply:
Great question, I'd need to do some modifications to iree_ref_utils to always enable this to true, but I was hoping we can do that in a follow-on PR. :)

Reply:
That works as well. Sure, we can save that for the next PR.